src/EditorFeatures/CSharpTest/Diagnostics/GenerateType/GenerateTypeTests.cs

```diff
@@ -140,6 +179,302 @@ internal class Base
 parameters: new TestParameters(Options.Regular));
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateType)]
+        public async Task TestGenerateInternalInterfaceFromASingleConstraintClause()
+        {
+            await TestInRegularAndScriptAsync(
+@"interface IEmployeeList<T> where T : [|Employee|], new()
+{
+}",
+@"interface IEmployeeList<T> where T : Employee, new()
+{
+}
+
+internal class Employee
+{
+}",
+index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateType)]
+        public async Task TestGeneratePublicInterfaceFromASingleConstraintClause()
```

Comment:
:bulb: Another good test would be the following. Unlike other types, if you omit the accessibility keyword from an interface member (which is actually a requirement in this location), the result is effectively `pubilc` instead of `private`.

```csharp
public interface IFoo
{
    void GetSomething<T>() where T : D;
}
```